multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
+            
+            patch_end = k[i+5].split(':',1)[1].strip()
+            patch_end = datetime.strptime(patch_end, "%Y-%m-%d %H:%M:%S.%f")
+            patch_end = time.mktime(patch_end.timetuple())
+            
+            pipette_log[k[i].strip(':')[0]] = {
+                        'pipette_status': k[i+1].split(':')[-1],
+                        'got_data':  k[i+2].split(':')[-1].strip(),
+                        'ad_channel': k[i+3].split(':')[-1].strip(),
+                        'patch_start': patch_start,
+                        'patch_end': patch_end,
+                        'cell_labels': 
+                            {
+                            'biocytin': k[i+7].split(':')[-1].strip(),
+                            'blue': k[i+8].split(':')[-1].strip(),
+                            'green': k[i+9].split(':')[-1].strip(),
+                            'red': k[i+10].split(':')[-1].strip()
+                            },
+                        'target_layer': k[i+11].split(':')[-1].strip(),
+                        'morphology': k[i+12].split(':')[-1].strip(), 
+                        'internal_solution': k[i+13].split(':')[-1].strip(),
+                        'internal_dye': k[i+14].split(':')[-1].strip(),
+                        'synapse_to': k[i+15].split(':')[-1].strip(),
+                        'gap_to': k[i+16].split(':')[-1].strip(),
+                        'notes': k[i+17].split(':')[-1].strip()
+                        }   
+
+        pipettes = markers[0].saveState()['markers']
+
+        if len(pipettes) != len(pipette_log):
+            raise Exception("There are a different number of pipettes in the pipettes.yml ")
+        for cell in pipettes:
+            p = cell[1]
+            
+            #print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).x())
+            #print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).y())
+        data = {}  
+        data['cells'] = []
+        for cell in pipettes:
+            cell_name = str(int(cell[0].split('_')[-1]))
+            p = cell[1]
+            data['cells'].append({      
+                'ephys_cell_id': cell_name,                  
+                'coordinates_20x': 
+                {
+                "x": markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).x(), 
+                "y": markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).y()
+                },      
+                'ephys_qc_result': ('pass' if pipette_log[cell_name]['got_data'] == 'true' else 'fail'),                 
+                'start_time_sec': pipette_log[cell_name]['patch_start']                     
+            })
+        
+        # as defined by requirements from technology
+
+        json_save_file = cluster_name + "_ephys_cell_cluster.json"
+        print (json_save_file)
+
+        if lims.is_mouse(self.slice_name) == 'mouse':
```

Comment:
I think you're actually supposed to read the incoming path from lims because they might want to change it in the future.

Reply:
I have not seen a place in LIMS for this and I think the incoming folders are outside